### PR TITLE
[Issue #1356] Create more subnets for MicroHealth

### LINKED ISSUE #1356: [Task]: Create a security subnet to host MicroHealth tooling
### Summary

Requested by Shane Stecker

### Acceptance criteria

_No response_

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary
Fixes #1356

## Changes proposed

Creates a set of "security subnets" by MicroHealth's request

## Testing

I deployed these changes to `dev` to test

### CODE REVIEW

File: infra/modules/network/main.tf [hunk 1]
```diff
@@ -35,3 +35,65 @@ module "aws_vpc" {
...
+
+# The configuration below creates "security subnets" with are used exclusively for
+# MicroHealth's security tooling. They were created by MicroHealth's request to confirm
+# to their standard architecture.
+
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/subnet
+resource "aws_subnet" "security_public" {
+  vpc_id                  = module.aws_vpc.vpc_id
+  availability_zone       = local.availability_zones[0]
+  cidr_block              = "10.${var.second_octet}.13.0/24"
```
Comment (jamesbursa) [inline infra/modules/network/main.tf:51]: Above we used third octets 0 to 2, 5 to 7, and 10 to 12. Should we keep that pattern of 5s and use 15 here (and 16 below)?

That way if for example we switch to 4 azs we'll be able to keep the subnets of each kind together.

**Review by jamesbursa — APPROVED** [review]
Looks good, thanks!

File: infra/modules/network/main.tf (hunk 1 above)
Comment (coilysiren) [reply infra/modules/network/main.tf:51]: Oh, nice catch! Changing that now

Comment (coilysiren) [reply infra/modules/network/main.tf]: Didn't work. I got this:

> InvalidSubnet.Range: The CIDR '10.0.16.0/24' is invalid.

Comment (coilysiren) [reply infra/modules/network/main.tf]: OH! The VPC cidr is only `10.0.0.0/20`. That's... small. Probably my own fault 🤷🏼 but too late to do anything about it now.

**Pull request merged**